ocaml/typing/typecore.ml

```diff
@@ -6491,7 +6518,7 @@ and type_application env app_loc expected_mode (position, maybe_outer_ret_mode)
           (Value_mode.regional_to_local_alloc funct_mode) sargs ret_tvar
       in
       let partial_app = is_partial_apply untyped_args in
-      let position = if partial_app then Default else position in
+      let position = if partial_app then Default else pm.apply_position in
```

Comment:
Rather than checking `partial_app` twice in this function, this should just return a whole `pm`.